implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -70,19 +70,29 @@ public DynatraceExporterV2(DynatraceConfig config, Clock clock, HttpSender httpC
         metricBuilderFactory = factoryBuilder.build();
     }
 
-    private String prepareEndpoint(DynatraceConfig config) {
-        String endpoint;
-
-        if (config.uri().matches("/metrics/ingest/?$")) {
-            endpoint = config.uri();
-        } else {
-            try {
-                endpoint = new URL(new URL(config.uri()), "/api/v2/metrics/ingest").toString();
-            } catch (MalformedURLException e) {
-                logger.warn("could not parse endpoint url. Falling back to local OneAgent endpoint.");
-                endpoint = "http://127.0.0.1:14499/metrics/ingest";
+    String prepareEndpoint(String uri) {
+        String endpoint = DEFAULT_ONEAGENT_ENDPOINT;
+
+        if (!uri.isEmpty()) {
+            // ends with "/metrics/ingest" or "/metrics/ingest/"
+            if (uri.matches(".+/metrics/ingest/?$")) {
+                endpoint = uri;
+            } else {
+                try {
+                    if (uri.contains("localhost") || uri.contains("127.0.0.1")) {
+                        // append /metrics/ingest for local endpoints if not already there.
+                        endpoint = new URL(new URL(uri), "/metrics/ingest").toString();
```

Comment:
This will fail for `::1` an so on but I'm not sure if the local endpoint even supports IPv6.